bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -203,6 +207,8 @@ fn build_thumbnail_info(
 pub struct UploadParameters {
     /// Filename (previously called "url") for the media to be sent.
     filename: String,
+    /// Optional file data - if not present, then data is read from `filename`
```

Comment:
So this might be slightly more idiomatic as a new enum with a `Filename` and `Data` option.  It would be a 'breaking' change for the API, but is more in keeping with the way the FFI is generally structured.

Reply:
I'm definitely open to it! I partly avoided this because wrapping `AttachmentSource` seemed a bit esoteric and duplicating it seemed a bit crude. Of course, what I have here is also rather crude 😅

Reply:
I agree that the enum is the better approach, can we have that please?

Reply:
No problem, I will update it!

Reply:
Thanks, and sorry for the delay, somehow this PR fell through the cracks.